x/crosschain/simulation/operation_vote_inbound.go

```diff
@@ -0,0 +1,190 @@
+		// Fetch the account from the auth keeper which can then be used to fetch spendable coins
+		authAccount := k.GetAuthKeeper().GetAccount(ctx, simAccount.Address)
+		spendable := k.GetBankKeeper().SpendableCoins(ctx, authAccount.GetAddress())
+
+		// Generate a transaction with a random fee and deliver it
+		txCtx := simulation.OperationInput{
+			R:               r,
+			App:             app,
+			TxGen:           moduletestutil.MakeTestEncodingConfig().TxConfig,
+			Cdc:             nil,
+			Msg:             &msg,
+			MsgType:         msg.Type(),
+			Context:         ctx,
+			SimAccount:      simAccount,
+			AccountKeeper:   k.GetAuthKeeper(),
+			Bankkeeper:      k.GetBankKeeper(),
+			ModuleName:      types.ModuleName,
+			CoinsSpentInMsg: spendable,
+		}
+
+		// Generate and deliver the transaction using the function defined by us instead of using the default function provided by the cosmos-sdk
+		// The main difference between the two functions is that the one defined by us does not error out if the vote fails.
+		// We need this behaviour as the votes are assigned to future operations, i.e., they are scheduled to be executed in a future block. We do not know at the time of scheduling if the vote will be successful or not.
+		// There might be multiple reasons for a vote to fail , like the observer not being present in the observer set, the observer not being an observer, etc.
+		return GenAndDeliverTxWithRandFees(txCtx)
+	}
+}
+
+// SimulateVoteInbound generates a MsgVoteInbound with random values and delivers it. It also schedules future operations for subsequent votes.
+func SimulateVoteInbound(k keeper.Keeper) simtypes.Operation {
+	observerVotesTransitionMatrix, statePercentageArray, curNumVotesState := ObserverVotesSimulationMatrix()
+	return func(
+		r *rand.Rand,
+		app *baseapp.BaseApp,
+		ctx sdk.Context,
+		accs []simtypes.Account,
+		chainID string,
+	) (simtypes.OperationMsg, []simtypes.FutureOperation, error) {
+		to, from := int64(1337), int64(101)
+		supportedChains := k.GetObserverKeeper().GetSupportedChains(ctx)
+		for _, chain := range supportedChains {
+			if chains.IsEthereumChain(chain.ChainId, []chains.Chain{}) {
+				from = chain.ChainId
+			}
+			if chains.IsZetaChain(chain.ChainId, []chains.Chain{}) {
+				to = chain.ChainId
+			}
+		}
+
+		asset, err := GetAsset(ctx, k.GetFungibleKeeper(), from)
+		if err != nil {
+			return simtypes.NoOpMsg(types.ModuleName, authz.InboundVoter.String(), "unable to get asset"), nil, err
+		}
+
+		// Generate a random inbound vote , coin type is randomly selected
+		msg := sample.InboundVoteFromRand(from, to, r, asset)
+
+		cf, found := k.GetObserverKeeper().GetCrosschainFlags(ctx)
+		if !found {
+			return simtypes.NoOpMsg(types.ModuleName, msg.Type(), "crosschain flags not found"), nil, nil
+		}
+
+		// Return early if inbound is not enabled.
+		if !cf.IsInboundEnabled {
+			return simtypes.NoOpMsg(types.ModuleName, msg.Type(), "inbound is not enabled"), nil, nil
+		}
+
```

Comment:
Added this check in this PR, which would fail this transaction if inbound is disabled , instead of erroring when delivering the message( which causes the the test to stop )

This would allow us to reduce the weight we used for MsgEnableCCTX , without causing the tests to break . Add the cost of increased failures for `VoteInbound` 

This comment thread from the last pr for reference 
https://github.com/zeta-chain/node/pull/3095#discussion_r1841447204

However, as mentioned in the comment thread, I think adding a flag would be the best way forward